lib/rubocop/cop/rails/inexplicit_filter.rb

```diff
@@ -0,0 +1,113 @@
+      # This cop checks that methods specified in the filter's `only`
+      # or `except` options are explicitly defined in the controller.
+      #
+      # You can specify methods of superclass or methods added by mixins
+      # on the filter, but these confuse developers. If you specify methods
+      # where are defined on another controller, you should define the filter
+      # in that controller.
+      #
+      # @example
+      #   # bad
+      #   class LoginController < ApplicationController
+      #     before_action :require_login, only: %i[index settings logout]
+      #
+      #     def index
+      #     end
+      #   end
+      #
+      #   # good
+      #   class LoginController < ApplicationController
+      #     before_action :require_login, only: %i[index settings logout]
+      #
+      #     def index
+      #     end
+      #
+      #     def settings
+      #     end
+      #
+      #     def logout
+      #     end
+      #   end
+      class InexplicitFilter < Cop
+        MSG = '%s not explicitly defined on the controller.'.freeze
+
+        FILTERS = %w[
+          :after_action
+          :append_after_action
+          :append_around_action
+          :append_before_action
+          :around_action
+          :before_action
+          :prepend_after_action
+          :prepend_around_action
+          :prepend_before_action
+          :skip_after_action
+          :skip_around_action
+          :skip_before_action
+          :skip_action_callback
+        ].freeze
+
+        def_node_matcher :only_or_except_filter_methods, <<-PATTERN
+          (send
+            nil?
+            {#{FILTERS.join(' ')}}
+            _
+            (hash
+              (pair
+                (sym {:only :except})
+                $_)))
+        PATTERN
+
+        def on_send(node)
+          methods_node = only_or_except_filter_methods(node)
+          return unless methods_node
+
+          methods = array_values(methods_node)
+          defined_methods = []
+          node.parent.each_child_node(:def) do |def_node|
+            defined_methods << def_node.method_name
+          end
+
+          @inexplicit_method_names = []
+          methods.each do |method|
+            unless defined_methods.include?(method)
+              @inexplicit_method_names << method
+            end
+          end
+
+          add_offense(node) unless @inexplicit_method_names.empty?
+        end
+
+        def message(_node)
+          if @inexplicit_method_names.size == 1
+            format(MSG, "`#{@inexplicit_method_names[0]}` is")
+          else
+            format(MSG, "`#{@inexplicit_method_names.join('`, `')}` are")
+          end
+        end
+
+        private
+
+        def array_values(node)
+          case node.type
+          when :str
+            [node.to_a[0]]
+          when :sym
+            [node.value]
+          when :array
+            node.values.map do |v|
+              v.instance_of?(RuboCop::AST::SymbolNode) ? v.value : v
```

Comment:
I think it should be like the following.

```ruby
node.values.map do |v|
  case v.type
  when :sym
    v.value
  when :str
    v.str_content
  end
end.compact
```

I think the cop should ignore not-literal value.